selfservice/strategy/password/validator.go

```diff
@@ -121,6 +121,7 @@ func (s *DefaultPasswordValidator) fetch(hpw []byte, apiDNSName string) error {
 	}
 
 	s.hashes.SetWithTTL(b20(hpw), 0, 1, hashCacheItemTTL)
+	s.hashes.Wait()
```

Comment:
@harnash without this wait, the correct `SetWithTTL` later on would be in a race condition with this one, causing the validator to mistakenly think that a password has not leaked. The wait ensures that the cache had enough time to consolidate the state, and any subsequent cache sets will override the items